pkg/kubectl/cmd/proxy.go

```diff
@@ -83,14 +86,68 @@ func NewCmdProxy(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 	cmd.Flags().StringP("address", "", "127.0.0.1", "The IP address on which to serve on.")
 	cmd.Flags().Bool("disable-filter", false, "If true, disable request filtering in the proxy. This is dangerous, and can leave you vulnerable to XSRF attacks, when used with an accessible port.")
 	cmd.Flags().StringP("unix-socket", "u", "", "Unix socket on which to run the proxy.")
+	cmd.Flags().StringP("exec", "e", "", "Command to be execured with the proxy address.")
 	return cmd
 }
 
+type executorInterface interface {
+	Exec(argv0 string, argv []string) error
+	LookPath(argv []string) (string, error)
+}
+
+type syscallExecutor struct{}
+
+func (*syscallExecutor) Exec(argv0 string, argv []string) error {
+	cmd := exec.Command(argv0, argv...)
+	cmd.Stdout = os.Stdout
+	cmd.Stdin = os.Stdin
+
+	if err := cmd.Run(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (*syscallExecutor) LookPath(argv []string) (string, error) {
+	return exec.LookPath(argv[0])
+}
+
+type proxyExec struct {
+	executor executorInterface
+}
+
+func (p proxyExec) Run(execCommand, address string, port int) error {
+	r, err := regexp.Compile("{}")
+	if err != nil {
+		return err
+	}
+	execCommand = r.ReplaceAllString(execCommand, fmt.Sprintf("http://%s:%d", address, port))
+
+	argv := strings.Split(execCommand, " ")
```

Comment:
Does this work if my command has empty spaces? For example, what if the command I want to run is `--exec="echo 'foo bar'"`?